core/services/ocr2/plugins/directrequestocr/reporting.go

```diff
@@ -360,7 +407,12 @@ func (r *functionsReporting) ShouldTransmitAcceptedReport(ctx context.Context, t
 		"needTransmissionIds": needTransmissionIds,
 		"transmitting":        len(needTransmissionIds) > 0,
 	})
-	return len(needTransmissionIds) > 0, nil
+	shouldTransmit := len(needTransmissionIds) > 0
+	if shouldTransmit {
+		promReportingTransmitReports.WithLabelValues(r.jobID.String()).Inc()
+		promReportingTransmitBatchSize.WithLabelValues(r.jobID.String()).Add(float64(len(needTransmissionIds)))
```

Comment:
needTransmissionIds can be smaller than the actual batch size. You should use len(allIds) here.